gcip/addons/kaniko/jobs.py

```diff
@@ -67,7 +67,7 @@ def execute(
             Defaults to `None` which skips the docker login attempt.
 
     Returns:
-        Job: gcip.Job will be returned to create container images.
+        Job: gcip.Job will be returned to create container images with ```name=kaniko``` and ```namespace=execute```.
```

Comment:
Is this intended? You explained, that Job name and/or namespace should be specified by the user.

Reply:
Yes, this is intended. Every job/sequence needs a name and/or namespace to be given. Else an exception would be raised.

Recently I explained you something different. Assume following function:

```
get_special_job(upload_target: str) -> Job:
  return Job(namespace="upload", script=f"upload --to {upload_target}")
```

The user now can control whether he wants to run the jobs in parallel...

```
pipeline.add_children(get_special_job("foo"), name="foo")
pipeline.add_children(get_special_job("bar"), name="bar")

# stages:
# - upload
# upload-foo:
#   stage: upload
# upload-bar:
#   stage: upload
```

... or in sequence...

```
pipeline.add_children(get_special_job("foo"), namespace="foo")
pipeline.add_children(get_special_job("bar"), namespace="bar")

# stages:
# - upload_foo
# - upload_bar
# upload-foo:
#   stage: upload_foo
# upload-bar:
#   stage: upload_bar
```

Recently you presented me another design:

```
get_special_job(upload_target: str) -> Job:
  return Job(namespace=upload_target, script=f"upload --to {upload_target}")
```

The differene here was, that you named the function after an argument of the function. Here you force to run all jobs in sequence, because for every different argument `upload_target` the job will be in a separate stage named after the `upload_target` argument.

Yes, you could avoid the problem by presetting `name` instead of `namespace`:

```
get_special_job(upload_target: str) -> Job:
  return Job(name=upload_target, script=f"upload --to {upload_target}")
```

But this design assumed, that the user does not need to set a name or namespace. Because you say that your job already sets the name(/namespace) to a distinct value, you dear user does not add the name/namespace "foo" or "bar" again:

```
pipeline.add_children(get_special_job("foo"))
pipeline.add_children(get_special_job("bar"))

# stages:
# - test
# test-foo:
#   stage: test
# test-bar:
#   stage: test
```

... because then in this case it is useless for the user to set the name when adding then job:

```
pipeline.add_children(get_special_job("foo"), name="foo")
pipeline.add_children(get_special_job("bar"), name="bar")

# stages:
# - test
# test-foo-foo:
#   stage: test
# test-bar-bar:
#   stage: test
```

And here I argued that this is bad design. Because how does the user know? In this case he must consult the documentation or code of every job-returning function to check whether the function already sets a distinct name/namespace of the returned job or not.

What I tried to explain was, that the user should care for a distinct name or namespac when adding a job/sequence.

That means - a job returning function should (technically has to) set the name and/or namespace of the job, but to a static value. The user should never assume that two jobs have distinct names because of the arguments of that job returning function and always set name/namespace with a distinct value himself.

Reply:
Ahh, thx for your brief explanation.